### PR TITLE
KAFKA-8034: Use automatic RPC generation in DeleteTopics

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mimaison** [comment]
retest this please

**Comment by mimaison** [comment]
@omkreddy @cmccabe If you have some time, can you take a look? Thanks

File: clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java [hunk 1]
```diff
@@ -62,85 +59,62 @@
...
-                return new DeleteTopicsResponse(topicErrors);
-            case 1:
-            case 2:
-            case 3:
-                return new DeleteTopicsResponse(throttleTimeMs, topicErrors);
-            default:
-                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
-                    version(), this.getClass().getSimpleName(), ApiKeys.DELETE_TOPICS.latestVersion()));
+        DeleteTopicsResponseData response = new DeleteTopicsResponseData();
+        if (version >= 3) {
```
Comment (cmccabe) [inline clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java]: Should be `if (version >= 1) {` to match the previous code, right?

**Comment by cmccabe** [comment]
Thanks, @mimaison!  Looks good overall.  I left one correction

File: clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java (hunk 1 above)
Comment (mimaison) [reply clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java]: Good catch! I've just pushed a fix

**Comment by cmccabe** [comment]
LGTM.  Thanks, @mimaison

**Pull request merged**